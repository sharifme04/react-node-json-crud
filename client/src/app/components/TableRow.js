import React from "react";
import { useDispatch } from "react-redux";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { deleteToDo } from "../actions/todo";

const TableRow = ({ e }) => {
  const dispatch = useDispatch();
  return (
    <tr>
      <td>{e?.id}</td>
      <td>{e?.title}</td>
      <td>{e?.description}</td>
      <td>{e?.date}</td>
      <td>
        <Link to={`/todo/edit/${e?.id}`}>
          <button type="button" className="btn btn-info">
            Edit
          </button>
        </Link>
      </td>
      <td>
        <button
          type="button"
          className="btn btn-danger"
          onClick={() => dispatch(deleteToDo(e?.id))}
        >
          Delete
        </button>
      </td>
    </tr>
  );
};

TableRow.propTypes = {
  e: PropTypes.object,
};

export default TableRow;
