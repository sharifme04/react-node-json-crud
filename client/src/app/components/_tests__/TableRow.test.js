import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { MemoryRouter } from "react-router-dom";
import TableRow from "../TableRow";

const mockStore = configureMockStore();
const store = mockStore({});

const tableRow = {
  title: "sharif 954t5",
  description: "This is a jkn5lkt text fosdmg, gmrat",
  id: 8422553,
  date: "17-08-2022",
};

test("snapshot SearchField renders with empty", () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <TableRow />
      </MemoryRouter>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot SearchField renders with empty props", () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <TableRow key={""} e={{}} />
      </MemoryRouter>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot SearchField renders with props", () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <TableRow key={"keyid"} e={tableRow} />
      </MemoryRouter>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
