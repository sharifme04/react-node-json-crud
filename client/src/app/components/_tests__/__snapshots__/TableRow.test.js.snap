// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot SearchField renders with empty 1`] = `
<tr>
  <td />
  <td />
  <td />
  <td />
  <td>
    <a
      href="/todo/edit/undefined"
      onClick={[Function]}
    >
      <button
        className="btn btn-info"
        type="button"
      >
        Edit
      </button>
    </a>
  </td>
  <td>
    <button
      className="btn btn-danger"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </td>
</tr>
`;

exports[`snapshot SearchField renders with empty props 1`] = `
<tr>
  <td />
  <td />
  <td />
  <td />
  <td>
    <a
      href="/todo/edit/undefined"
      onClick={[Function]}
    >
      <button
        className="btn btn-info"
        type="button"
      >
        Edit
      </button>
    </a>
  </td>
  <td>
    <button
      className="btn btn-danger"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </td>
</tr>
`;

exports[`snapshot SearchField renders with props 1`] = `
<tr>
  <td>
    8422553
  </td>
  <td>
    sharif 954t5
  </td>
  <td>
    This is a jkn5lkt text fosdmg, gmrat
  </td>
  <td>
    17-08-2022
  </td>
  <td>
    <a
      href="/todo/edit/8422553"
      onClick={[Function]}
    >
      <button
        className="btn btn-info"
        type="button"
      >
        Edit
      </button>
    </a>
  </td>
  <td>
    <button
      className="btn btn-danger"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </td>
</tr>
`;
